modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/SynonymTokenFilterFactory.java

```diff
@@ -97,7 +100,7 @@ public String name() {
 
             @Override
             public TokenStream create(TokenStream tokenStream) {
-                return synonyms.fst == null ? tokenStream : new SynonymFilter(tokenStream, synonyms, false);
+                return synonyms.fst == null ? tokenStream : new BoostableSynonymFilter(tokenStream, synonyms, false, boost);
```

Comment:
Note: I didn't find an easy way to extend or overwrite the existing filters behaviour, since the whole logic for adding tokens essentially sits in `incrementToken()` which doesn't provide any specific hook for adding attributed to tokens. I consider this copied a hopefully temporary workaround.

Reply:
Can be solved by wrapping the existing filter into another one that adds the boost factor to all tokens with a specific type (e.g. Synonym type)